content/inside-rust/stage0-redesign/index.md

```diff
@@ -0,0 +1,126 @@
++++
+path = "inside-rust/9999/12/31/redesigning-the-initial-bootstrap-sequence"
+title = "Redesigning the Initial Bootstrap Sequence"
+authors = ["Jieyou Xu"]
+
+[extra]
+team = "the Bootstrap team"
+team_url = "https://www.rust-lang.org/governance/teams/infra#team-bootstrap"
++++
+
+This blog post accompanies an [upcoming major change to the `bootstrap` build system][stage0-redesign-pr] (see also [Major Change Proposal 619][redesign-stage0-mcp]), which is expected to affect contributors who works on the `rustc` compiler, the standard library, tools, and the distributors of rust toolchains.
+
+# TL;DR: What is being changed?
+
+We are [redesigning the stage 0 bootstrap sequence][stage0-redesign-pr] so that the standard library will now only support one compiler version. Building the stage 1 compiler will no longer involve building the in-tree standard library. Instead, the pre-built beta standard library will be used.
```

Comment:
I'm not sure if I would use "stage 0" or "stage 1" in the TL;DR section.

One thought is that explaining this by providing the x.py build log before and after the change, or some similarly visual-ish comparison might be clearer than trying to *name* things (e.g., "Building the stage 1 compiler ..." is probably ambiguous to the average reader, I think you mean the compiler that ends up in build/host/stage1/bin/rustc, but that's in some sense the 0th compiler we build...).

Reply:
In some sense, yeah, this wording is... hard.

Reply:
FWIW having seen the diagrams... it might be OK to just have those shown here? Maybe with the exact text that x.py prints prior to running that section of the build?

Those seem like a very nice before/after comparison to me...

Reply:
I might drop the "stage 0" vs "stage 1 " here, and instead use "beta" and "in-tree" instead, *even though* this is not technically always the case (e.g. stage 0 doesn't have to be beta). I can move the "simplifying assumption" footnote earlier to here.